packages/grafana-data/src/transformations/transformers/fillGaps.ts

```diff
@@ -0,0 +1,61 @@
+import { MutableDataFrame, DataFrame, FieldType } from '../..';
+
+import { map } from 'rxjs/operators';
+
+import { DataTransformerID } from './ids';
+import { DataTransformerInfo } from '../../types/transformations';
+
+export interface FillGapsTransformerOptions {
+  interval?: number;
+  fillWith: any;
+}
+
+export const fillGapsTransformer: DataTransformerInfo<FillGapsTransformerOptions> = {
+  id: DataTransformerID.fillGaps,
+  name: 'Fill Gaps',
+  description: '',
+  defaultOptions: {
+    fillWith: null,
+  },
+  operator: (options) => (source) =>
+    source.pipe(
+      map((dataFrames) => {
+        if (!Array.isArray(dataFrames)) {
+          return dataFrames;
+        }
+        return dataFrames.map((f) => fillGaps(f, options));
+      })
+    ),
+};
+
+const getInterval = (frame: DataFrame) => frame.fields.map((f) => f.config.interval ?? 0).find((i) => i > 0) ?? 0;
+
+export const fillGaps = (df: DataFrame, options?: FillGapsTransformerOptions): DataFrame => {
+  const frame = new MutableDataFrame(df);
+  const interval = options?.interval ?? getInterval(frame);
+  if (interval === 0) {
+    return frame;
+  }
```

Comment:
hmm... what do you think of an alternative algortithm? 
- we would do it so that the interval is on the time-field itself

1. we first filter out all time-fields (to handle the multiple-time-fields case)
2. if there are multiple time fields, return error, or just skip transform
3. we check the remaining single time-field for interval. we use that interval

Reply:
this behavior has been changed to support number and time fields with intervals set